runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/IJobManagerTest.java

```diff
@@ -1893,11 +1913,11 @@ public void testSleep() {
 		assertState("2.0", job, Job.RUNNING);
 		assertTrue("2.1", !job.sleep());
 		assertState("2.2", job, Job.RUNNING);
-
+		job.terminate();
```

Comment:
All explicit terminations of jobs (i.e. `TestJob.terminate`) contribute to the fix mentioned in the commit message.
The rest of the changes are a pure refactoring to make explicit that the random numbers for job execution time have no meaning but simply a long running job is created that will be manually terminated at a specific point in time. Since some of previously short running jobs have to be made long running, this refactoring should improve comprehensibility in contrast to simply improving the execution time of the job.

Reply:
I see that combining the refactoring with the actual fix makes reviewing harder. However, having a PR only for such a simple refactoring (which I really think is useful) does not make sense to me either. Wouldn't it be an option to have separate commits in a PR, one with the actual fix and one with a related refactoring? Would ease reviewing but still provides a lightweight way to integrate reasonable refactorings.

Reply:
> I see that combining the refactoring with the actual fix makes reviewing harder. However, having a PR only for such a simple refactoring (which I really think is useful) does not make sense to me either. Wouldn't it be an option to have separate commits in a PR, one with the actual fix and one with a related refactoring? Would ease reviewing but still provides a lightweight way to integrate reasonable refactorings.

If possible prefer separate commits and PRs for refactorings.

Reply:
Please single commits per PR (unless something like added Version updates). Totally fine if a fix only contains a single line if it completely solves an issue. The refactoring seems unrelated.

Reply:
I have extracted the refactoring from this PR, so only the actual fixes for the issue remain.

Reply:
I still don't get it: how do you thing changing timing makes anything deterministic? A deterministic test would be independent of any time value.

Reply:
The changes explicitly terminate the jobs when they are expected to terminate instead of hoping that they will run long enough because of giving them enough execution time. The jobs will basically run indefinitely unless they are explicitly terminated. That is what makes their execution deterministic.
Of course, if you have an infrastructure on which execution exceeds the extremly high execution time values still attached to them (multiple hours), the tests will still fail. But I do assume this to be a realistic case to consider.